CMakeLists.txt

```diff
@@ -342,6 +342,25 @@ macro(find_jemalloc)
     endif()
 endmacro(find_jemalloc)
 
+macro(find_mimalloc)
+    if(NOT TARGET mimalloc)
+    ExternalProject_Add(libmimalloc
+        SOURCE_DIR "${PROJECT_SOURCE_DIR}/thirdparty/mimalloc"
+        BUILD_ALWAYS 1
+        PATCH_COMMAND make -p out/release
+        BINARY_DIR "${PROJECT_SOURCE_DIR}/thirdparty/mimalloc/out/release"
+        BUILD_COMMAND cmake -DMI_OVERRIDE=OFF ../.. 
+        INSTALL_COMMAND ${CMAKE_MAKE_PROGRAM}
+    )
+    add_library(mimalloc STATIC IMPORTED)
+    add_dependencies(mimalloc libmimalloc)
+    set_target_properties(mimalloc PROPERTIES IMPORTED_LOCATION
+                                                  ${PROJECT_SOURCE_DIR}/thirdparty/mimalloc/out/release/libmimalloc.a)
+    set_target_properties(mimalloc PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
+                                              ${PROJECT_SOURCE_DIR}/thirdparty)
+    endif()
+endmacro(find_mimalloc)
```

Comment:
To include dependencies with cmake support, please use `add_subdirectory_static` macro.

Reply:
Got it.